CMakeLists.txt

```diff
@@ -77,7 +77,6 @@ include_directories(
     ${CMAKE_SOURCE_DIR}/proof
     ${CMAKE_SOURCE_DIR}/models
     ${CMAKE_SOURCE_DIR}/itehandler
-    ${GMP_INCLUDE_DIRS}
```

Comment:
A build in my OS X with `GMP` installed from homebrew fails with the current commit, but passes if I add here ${GMP_INCLUDE_DIR}.  I wonder if the same would happen in some systems with `libreadline` or `libedit`.  Would it be ok to add ${GMP_INCLUDE_DIR} here at least?

Reply:
Hmm, I would like NOT to do that. One of the points for these changes is that exactly things like that should not be necessary because it should be done automatically.
Where do you get a compilation error?

Reply:
I see, I was just reading the documentation of cmake and it looks like that would not be the right way to go.  Anyway, to answer your question, it's my home laptop, where I'm running Mac OS X 10.15.2 and the cmake build system finds (correctly) `/usr/local/include` as the GMP_INCLUDE_DIR.

Reply:
Hmm, at my laptop it was working and I have GMP installed in the same directory. But I think I have older version of Mac OS.
What does the compilation error say?

Reply:
I managed to make this compile by adding `include_directories(${GMP_INCLUDE_DIR})` to `cmake_modules/FindGMP.cmake:32`.  Do you know if this is the correct way?

Reply:
Here's the error.

```
Anttis-MacBook-Pro:build anttihyvarinen$ make
[  0%] Building CXX object src/proof/CMakeFiles/proof.dir/PGBuild.cc.o
In file included from /Users/anttihyvarinen/src/opensmt/src/proof/PGBuild.cc:20:
In file included from /Users/anttihyvarinen/src/opensmt/src/proof/PG.h:23:
/Users/anttihyvarinen/src/opensmt/src/common/Global.h:36:10: fatal error:
      'gmp.h' file not found
#include <gmp.h>
         ^~~~~~~
1 error generated.
make[2]: *** [src/proof/CMakeFiles/proof.dir/PGBuild.cc.o] Error 1
make[1]: *** [src/proof/CMakeFiles/proof.dir/all] Error 2
make: *** [all] Error 2
```

Reply:
What you suggests might be the right solution. 
Let me have a look at the whole problem again and if I don't find other solution soon, we can go with this one.

Do you think you could send me a log of running `make VERBOSE=1`?

Reply:
As discussed, this is now resolved.